v5/pivotal/aggregator.go

```diff
@@ -0,0 +1,224 @@
+package pivotal
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+)
+
+const aggregatorURL = "https://www.pivotaltracker.com/services/v5/aggregator"
+
+// PT doesn't allow more than around 15 urls in one aggregation request.
+// However, this was found by experimenting. There was no official
+// rate-limiting data in their docs.
+const perPage = 15 // GET requests per aggregation requests.
+
+// AggregatorService is used to wrap the client.
+type AggregatorService struct {
+	client *Client
+}
+
+type AggregationRequest struct {
+	url       string
+	projectID int
+	storyID   int
+}
+
+// Aggregation object stores the state of the aggregation.
+type Aggregation struct {
+	// Requests are the GET requests that are used by the aggregator.
+	requests []string
+
+	// Storing the response json along with the urls.
+	aggregatedResponse map[string]interface{}
+
+	// The service of the aggregation
+	service *AggregatorService
+}
+
+func (a *AggregatorService) GetBuilder() *Aggregation {
+	aggregation := Aggregation{
+		aggregatedResponse: make(map[string]interface{}),
+		service:            a,
+	}
+	return &aggregation
+}
+
+func newAggregatorService(client *Client) *AggregatorService {
+	return &AggregatorService{client}
+}
+
+// Adds the url for the story to the aggregation data.
+func (a *Aggregation) Story(projectID, storyID int) *Aggregation {
+	a.requests = append(a.requests, BuildStoryURL(projectID, storyID))
+	return a
+}
+
+func (a *Aggregation) StoryUsingStoryID(storyID int) *Aggregation {
+	a.requests = append(a.requests, BuildStoryURLOnlyUsingStoryID(storyID))
+	return a
+}
+
+// Stories adds the urls for the slice of story IDs.
+func (a *Aggregation) Stories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.Story(projectID, storyID)
+	}
+	return a
+}
+
+// Adds the url for the comments to the aggregation data.
+func (a *Aggregation) CommentsOfStory(projectID, storyID int) {
+	a.requests = append(a.requests, buildCommentsURL(projectID, storyID))
+}
+
+// Comments adds the requests for getting the comments of the stories in storiesToGet.
+func (a *Aggregation) CommentsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.CommentsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+// Adds the url for the reviews to the aggregation data.
+func (a *Aggregation) ReviewsOfStory(projectID, storyID int) *Aggregation {
+	a.requests = append(a.requests, buildReviewsURL(projectID, storyID))
+	return a
+}
+
+// Comments adds the requests for getting the comments of the stories in storiesToGet.
+func (a *Aggregation) ReviewsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.ReviewsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+func BuildStoryURLOnlyUsingStoryID(storyID int) string {
```

Comment:
Do these URL helpers need to be public? I think they would only be used internally.